apps/server/src/modules/oauth/controller/api-test/oauth-sso.api.spec.ts

```diff
@@ -83,42 +85,87 @@ describe('OAuth SSO Controller (API)', () => {
 		app = moduleRef.createNestApplication();
 		await app.init();
 		em = app.get(EntityManager);
-
-		const wellKnown = app.get(KeycloakAdministrationService).getWellKnownUrl();
-		axiosMock.onGet(wellKnown).reply(200, {
-			issuer: 'issuer',
-			token_endpoint: 'tokenEndpoint',
-			authorization_endpoint: 'authEndpoint',
-			end_session_endpoint: 'logoutEndpoint',
-			jwks_uri: 'jwksEndpoint',
-		});
 	});
 
 	afterAll(async () => {
-		await cleanupCollections(em);
 		await app.close();
 	});
 
+	afterEach(async () => {
+		await cleanupCollections(em);
+	});
+
 	const setup = async () => {
 		const externalUserId = 'externalUserId';
 		const system: System = systemFactory.withOauthConfig().buildWithId();
 		const school: School = schoolFactory.buildWithId({ systems: [system] });
 		const user: User = userFactory.buildWithId({ externalId: externalUserId, school });
 		const account: Account = accountFactory.buildWithId({ systemId: system.id, userId: user.id });
 
-		await em.persistAndFlush([system, user, school, account]);
+		const targetSystem: System = systemFactory
```

Comment:
The reason why we introcude a setup method it the make it explict for any of the when describes and that it only contain data and mocks that are related to this when senario. I really expect that all the lines not related to each senario in this file.

Please move the setup to each of the when senarios. If something is needed on all places it is fine to let a block like createXYZ();

describe('when...')
const setup = async() => {
// createXYZ();